app/views/Main.js

```diff
@@ -1,52 +1,65 @@
 import { useNavigation } from '@react-navigation/native';
 import React, { useCallback, useEffect, useState, useContext } from 'react';
 import { AppState, BackHandler } from 'react-native';
+import { openSettings } from 'react-native-permissions';
 
-import { isGPS } from '../COVIDSafePathsConfig';
 import { checkIntersect } from '../helpers/Intersect';
 import BackgroundTaskServices from '../services/BackgroundTaskService';
 import LocationServices from '../services/LocationService';
-import ExposureNotificationService from '../services/ExposureNotificationService';
 import { AllServicesOnScreen } from './main/AllServicesOn';
 import {
   TracingOffScreen,
   NotificationsOffScreen,
   SelectAuthorityScreen,
 } from './main/ServiceOffScreens';
 import PermissionsContext, { PermissionStatus } from '../PermissionsContext';
+import ExposureNotificationsContext from '../ExposureNotificationContext';
 
 import { useSelector } from 'react-redux';
 import selectedHealthcareAuthoritiesSelector from '../store/selectors/selectedHealthcareAuthoritiesSelector';
 import { useStatusBarEffect } from '../navigation';
 
-export const Main = () => {
-  const tracingService = isGPS ? LocationServices : ExposureNotificationService;
+export const BTMain = () => {
+  useStatusBarEffect('light-content');
+  const [isENAuthorizedAndEnabled, setIsENAuthorizedAndEnabled] = useState(
+    false,
+  );
+  const {
+    getIsENAuthorizedAndEnabled,
+    requestENAuthorization,
+    deviceStatus,
+  } = useContext(ExposureNotificationsContext);
+
+  useEffect(() => {
+    console.log('STATUS: ', deviceStatus);
+    setIsENAuthorizedAndEnabled(getIsENAuthorizedAndEnabled());
```

Comment:
this seems a little redundant to have the state in both the context and the component. could we keep it as a single source of truth and determine the `isEnabledAndAuthorized` on the fly?
something like:
```typescript
const isEnabled = getIsEnabledAndAuthorzed()

return isEnabled ? <AllServicesOnScreen /> : <AllServicesOnScreen />
```

Reply:
We tried to do this without local state, but the changes within the context were not triggering a re-render when navigating between tabs. I would love to have you look more closely at that to help me understand why.

Reply:
ah yeah, this has to do with the way react-navigation does things. basically screen components wont go through their update cycle in this situation.  A way to get around this without introducing redundant local state is to pass the data in as a prop to a new component, so when the prop value updates, there will be re-render of the component.

so something like:
```typescript

const MyScreen = () => {
  const { myState } = useContext(MyContext)
  
  return (
    <View>
        <MyComponentThatIWantToUpdate myProp={myState} />
    </View>
}
```

so the parent screen component wont go through an update cycle on myState updating, but the child component `MyComponentThatIWantToUpdate` will.

Kinda annoying but it leads to some separation from the component that is responsible for screen transitions and the component that is responsible for displaying content on a screen, which is helpful i suppose.

Reply:
Its also possible that the context is not the correct fit for this use case, though i wouldnt be surprised if other components on the tree than this screen will need access to this.